lnd_test.go

```diff
@@ -12009,6 +12009,97 @@ func testSendUpdateDisableChannel(net *lntest.NetworkHarness, t *harnessTest) {
 	mineBlocks(t, net, 1)
 }
 
+// testAbandonChannel abandones a channel and asserts that it is not
+// longer open and not in one of the pending closure states. It also
+// verifies that the abandoned channel is reported as closed with close
+// type 'abandoned'.
+func testAbandonChannel(net *lntest.NetworkHarness, t *harnessTest) {
+	timeout := time.Duration(time.Second * 5)
+	ctxb := context.Background()
+
+	// First establish a channel between Alice and Bob.
+	ctxt, _ := context.WithTimeout(ctxb, timeout)
+
+	channelParam := lntest.OpenChannelParams{
+		Amt:     maxBtcFundingAmount,
+		PushAmt: btcutil.Amount(100000),
+	}
+
+	chanPoint := openChannelAndAssert(
+		ctxt, t, net, net.Alice, net.Bob, channelParam)
+
+	// Wait for channel to be confirmed open.
+	ctxt, _ = context.WithTimeout(ctxb, time.Second*15)
+	err := net.Alice.WaitForNetworkChannelOpen(ctxt, chanPoint)
+	if err != nil {
+		t.Fatalf("alice didn't report channel: %v", err)
+	}
+	err = net.Bob.WaitForNetworkChannelOpen(ctxt, chanPoint)
+	if err != nil {
+		t.Fatalf("bob didn't report channel: %v", err)
+	}
+
+	// Send request to abandon channel.
+	abandonChannelRequest := &lnrpc.AbandonChannelRequest{
+		ChannelPoint: chanPoint,
+	}
+
+	ctxt, _ = context.WithTimeout(ctxb, timeout)
+	_, err = net.Alice.AbandonChannel(ctxt, abandonChannelRequest)
+
+	if err != nil {
+		t.Fatalf("unable to abandon channel: %v", err)
+	}
+
+	// Assert that channel in no longer open.
```

Comment:
Seeing this flake a bit on Travis, as a result, would recommend that the checks below be wrapped in a `WaitPredicate`.